### ISSUE TITLE
Linux Support

### ISSUE THREAD

**Issue by dvdvgt** [Cawotte/FlockingCPP#1]
Hi,
your project looks really cool. I tried compiling it on Linux (Manjaro 5.9.1-1) however I was greeted by an error when running `make`.
```
FlockingCPP/src/SFMLApp.cpp:9:10: fatal error: windows.h: No such file or directory
    9 | #include "windows.h"
      |          ^~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/flockingBoids.dir/build.make:134: CMakeFiles/flockingBoids.dir/src/SFMLApp.cpp.o] Error 1
```
I guess this is some Windows specific header. Is there way to still make it work?

**Comment by dvdvgt** [comment]
I did get it to compile by do some modifications:

---

In Boid.cpp I changed `float detectionRadiusSquared = std::powf(detectionRadius, 2.);` to *float detectionRadiusSquared = std::pow(detectionRadius, 2.);`

---

In ImGuiExtra.h I changed `void	PlotVar(const char* label, float value, float scale_min = FLT_MAX, float scale_max = FLT_MAX, size_t buffer_size = 120);` to `void	PlotVar(const char* label, float value, float scale_min = FLT_MAX, float scale_max = FLT_MAX, std::size_t buffer_size = 120);`

---

In SFMLApp.cpp I commented out
```
//#include "windows.h"
//#include "psapi.h"
```
as well as the `SFMLApp::drawPerformanceUI()` function since it included Windows specific declarations.

---

After these modifications I was able to compile it. However the configuration widget obviously no longer included the performance UI.
